src/main/java/app/linguistai/bmvp/service/gamification/LeaderboardService.java

```diff
@@ -0,0 +1,145 @@
+package app.linguistai.bmvp.service.gamification;
+
+import app.linguistai.bmvp.exception.NotFoundException;
+import app.linguistai.bmvp.exception.SomethingWentWrongException;
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.gamification.IXPRanking;
+import app.linguistai.bmvp.model.gamification.UserXP;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import app.linguistai.bmvp.repository.gamification.IUserXPRepository;
+import app.linguistai.bmvp.response.gamification.RLeaderboardXP;
+import app.linguistai.bmvp.response.gamification.RUserXPRanking;
+import app.linguistai.bmvp.consts.LeaderboardConsts;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class LeaderboardService {
+    private final IAccountRepository accountRepository;
+    private final IUserXPRepository xpRepository;
+
+    @Transactional
+    public RLeaderboardXP getTopUsersByExperience(String email, Integer page, int size) throws Exception {
+        try {
+            // Fetch the global XP ranking for the logged-in user
+            IXPRanking loggedUserRanking = xpRepository.findGlobalUserRankByEmail(email);
+            if (loggedUserRanking == null) {
```

Comment:
Just for future reference, you can set JPA methods to return `Optional<>` and then use the following syntax for better readability and maintainability:

For exceptions:
`UserStreak userStreak = JPA_repo_call().orElseThrow(() -> new NotFoundException("error"));`

For assignment:
`UserStreak userStreak = JPA_repo_call().orElse(new UserStreak());`

Reply:
Yes, changed it - much more readable, thanks!